contracts/lper/src/contract.rs

```diff
@@ -0,0 +1,166 @@
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{MessageInfo,  Response,
+     StdResult, Addr, DepsMut, Env, Binary, Deps, to_binary, SubMsg, WasmMsg, CosmosMsg, Coin, Uint128, Reply, 
+};
+use cw2::set_contract_version;
+
+use astroport::{pair::{ExecuteMsg::ProvideLiquidity, Cw20HookMsg}, asset::{Asset, AssetInfo}};
+use cw20::Cw20ReceiveMsg;
+
+use crate::{msg::{ExecuteMsg, InstantiateMsg, MigrateMsg, QueryMsg}, state::{HOLDER_ADDRESS, LP_POSITION}};
+
+use neutron_sdk::{
+    bindings::{
+        msg::{NeutronMsg},
+        query::{NeutronQuery},
+    },
+    NeutronResult,
+};
+
+use crate::state::{
+   CLOCK_ADDRESS, CONTRACT_STATE, ContractState,
+};
+
+
+const CONTRACT_NAME: &str = "crates.io:covenant-lper";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: InstantiateMsg,
+) -> NeutronResult<Response<NeutronMsg>> {
+    deps.api.debug("WASMDEBUG: instantiate");
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+
+    // TODO: validations
+    CLOCK_ADDRESS.save(deps.storage, &Addr::unchecked(msg.clock_address))?;
+    CONTRACT_STATE.save(deps.storage, &ContractState::Instantiated)?;
+    LP_POSITION.save(deps.storage, &msg.lp_position)?;
+    HOLDER_ADDRESS.save(deps.storage, &msg.holder_address)?;
+
+    Ok(Response::default())
+}
+
+#[entry_point]
+pub fn execute(
+    mut deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> NeutronResult<Response<NeutronMsg>> {
+    deps.api
+        .debug(format!("WASMDEBUG: execute: received msg: {:?}", msg).as_str());
+    match msg {
+        ExecuteMsg::Tick {} => try_tick(deps, env, info),
+        ExecuteMsg::WithdrawRewards {} => try_withdraw(deps, env, info),
+    }
+}
+
+
+fn try_tick(deps: DepsMut, env: Env, info: MessageInfo) -> NeutronResult<Response<NeutronMsg>> {
+    let current_state = CONTRACT_STATE.load(deps.storage)?;
+
+    match current_state {
+        ContractState::Instantiated => try_enter_lp_position(deps, env, info),
```

Comment:
Where are these states getting updated? I'm not seeing where you save the contract state.